File: lymoon-mobile/src/types/schedule.ts
export type DayBar = { day: string; opacity: number; isToday?: boolean };

export type ScheduleCategory = 'All' | 'Shift' | 'Event' | 'Personal';

export type ScheduleItem = {
  id: string;
  title: string;
  subtitle: string;          // locally computed — NOT from API
  hours: string;
  iconBg: string;
  days: DayBar[];
  scheduleType?: 'shift' | 'event' | 'personal';
  memberPermission?: 'manager_only' | 'full_collaboration';
  startWeek?: string;        // ISO Monday date, e.g. "2026-03-16"
  currentWeek?: string;      // ISO Monday date — latest active week
  description?: string;
  inviteCode?: string;
};

export type ShiftType = 'Morning' | 'Standard' | 'Afternoon' | 'Custom';

export type Shift = {
  id: string;
  employeeId: string;
  dayOfWeek: number;  // 0 = Mon … 6 = Sun
  startTime: string;  // "09:00"
  endTime: string;    // "13:00"
  shiftType: ShiftType;
};

export type Employee = {
  id: string;
  name: string;
  role: string;
  avatarInitials: string;
};

export type ScheduleDetail = ScheduleItem & {
  employees: Employee[];
  shifts: Shift[];
  weekStartDate: string;      // ISO date "2024-10-14" (always a Monday)
  currentUserRole: 'Manager' | 'Member';
};

File: Lymoon.API/Models/Schedule.cs
namespace Lymoon.API.Models;

public class Schedule
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    /// <summary>'shift' | 'event' | 'personal'</summary>
    public string ScheduleType { get; set; } = string.Empty;
    /// <summary>'manager_only' | 'full_collaboration'</summary>
    public string MemberPermission { get; set; } = string.Empty;
    /// <summary>6-character uppercase alphanumeric, unique</summary>
    public string InviteCode { get; set; } = string.Empty;
    public string IconBg { get; set; } = string.Empty;
    /// <summary>ISO Monday — week the schedule was created for</summary>
    public DateOnly StartWeek { get; set; }
    /// <summary>ISO Monday — latest active week (advances forward only)</summary>
    public DateOnly CurrentWeek { get; set; }
    public string CreatedById { get; set; } = string.Empty;

    public AppUser CreatedBy { get; set; } = null!;
    public ICollection<ScheduleMember> Members { get; set; } = new List<ScheduleMember>();
    public ICollection<Shift> Shifts { get; set; } = new List<Shift>();
    public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
}
